### PR TITLE
Added setting for AM PM. Fixed 12 hour not being respected

### PR DESCRIPTION
**Pull request by Jkitche**
## Changes
This PR fixes the offset issue with the time provider always returning 12 when the time is past 12 pm. Instead, this switches to uses the libs `isPM` function instead.

Additionally, it adds a specific setting for showing the AM/PM indicator.

## Testing
Enable and disable the setting for the indicator to test the visibility
For very thorough testing, ensure the time is valid in both AM and PM scenarios. I will provide screenshots for both.

## Misc
image

### CODE REVIEW

File: Info-Orbs/src/widgets/clockWidget.cpp [hunk 1]
```diff
@@ -17,7 +17,8 @@ void ClockWidget::setup() {
 }
 
 void ClockWidget::draw(bool force) {
-    bool showFirstDigit = FORMAT_24_HOUR || (!FORMAT_24_HOUR && m_hourSingle >= 10);
+    GlobalTime* time = GlobalTime::getInstance();
+    bool showFirstDigit = FORMAT_24_HOUR || (!FORMAT_24_HOUR && time->isPM());
```
Comment (ericthelin) [inline Info-Orbs/src/widgets/clockWidget.cpp:21]: this is wrong. I think the first version is correct

File: Info-Orbs/src/widgets/clockWidget.cpp [hunk 2]
```diff
@@ -49,11 +50,12 @@ void ClockWidget::draw(bool force) {
 }
 
 void ClockWidget::displayAmPm(uint32_t color) {
+    GlobalTime* time = GlobalTime::getInstance();
     m_manager.selectScreen(2);
     TFT_eSPI& display = m_manager.getDisplay();
     display.setTextSize(4);
     display.setTextColor(color, TFT_BLACK, true);
-    const String& am_pm = m_hourSingle >= 12 ? "AM" : "PM";
+    const String& am_pm = time->isPM() ? "PM" : "AM";
```
Comment (ericthelin) [inline Info-Orbs/src/widgets/clockWidget.cpp:58]: why `const String &` instead of just `String`

Comment (Jkitche) [reply Info-Orbs/src/widgets/clockWidget.cpp:58]: It was just to match the types of `display.drawString`

File: Info-Orbs/src/widgets/clockWidget.cpp (hunk 1 above)
Comment (Jkitche) [reply Info-Orbs/src/widgets/clockWidget.cpp:21]: Yup, you're right. Got my changes mixed up.

**Review by ericthelin — APPROVED** [review]

**Comment by ericthelin** [comment]
Tested with all cases that seem relevant (24h, 12h pm, 12h am, 12h not shown)

**Pull request merged**